src/CostMatrix.cpp

```diff
@@ -0,0 +1,63 @@
+//
+// Created by mikita on 17.07.19.
+//
+
+#include "CostMatrix.h"
+
+bool CostMatrix::readFromFile(const std::string &filename)
+{
+    std::ifstream inFile;
+
+    inFile.open(filename);
+    if (!inFile)
+    {
+        std::cout << "Unable to open file " << filename << std::endl;
+        return false;
+    }
+
+    inFile >> this->size;
+    std::vector<std::vector<int>> matrix(size, std::vector<int>(size, 0));
+
+    try
+    {
+        for (int i = 0; i < this->size; ++i)
+        {
+            for (int j = 0; j < this->size; ++j)
+            {
+                inFile >> matrix[i][j];
+            }
+        }
+    }
+    catch (std::exception e)
+    {
+        std::cout << "File " << filename << " has bad format, unable to parse matrix" << std::endl;
+        return false;
+    }
+    costMatrix = std::move(matrix);
+    return true;
```

Comment:
it's generally better to implement methods returning new objects instead of modyfing existing ones.
Why not simply return created matrix?

Reply:
changed `readFromFile` method to stand-alone function, which return `CostMatrix`